python/kserve/kserve/protocol/rest/server.py

```diff
@@ -71,175 +58,60 @@ def install_signal_handlers(self) -> None:
-            exception_handlers={
-                InvalidInput: invalid_input_handler,
-                InferenceError: inference_error_handler,
-                ModelNotFound: model_not_found_handler,
-                ModelNotReady: model_not_ready_handler,
-                NotImplementedError: not_implemented_error_handler,
-                Exception: generic_exception_handler,
-            },
-        )
-        # Register OpenAI endpoints if any of the models in the registry implement the OpenAI inferface
+    def create_application(self):
+        """Create a KServe ModelServer application with API routes."""
+        root_router = APIRouter()
+        root_router.add_api_route(r"/", self.dataplane.live)
+        root_router.add_api_route(r"/metrics", metrics_handler, methods=["GET"])
+        self.app.include_router(root_router)
+        register_v1_endpoints(self.app, self.dataplane, self.model_repository_extension)
+        register_v2_endpoints(self.app, self.dataplane, self.model_repository_extension)
+        # Register OpenAI endpoints if any of the models in the registry implement the OpenAI interface
         # This adds /openai/v1/completions and /openai/v1/chat/completions routes to the
         # REST server.
-        maybe_register_openai_endpoints(app, self.dataplane.model_registry)
-        return app
+        maybe_register_openai_endpoints(self.app, self.dataplane.model_registry)
+
+        # Add exception handlers
+        self.app.add_exception_handler(InvalidInput, invalid_input_handler)
+        self.app.add_exception_handler(InferenceError, inference_error_handler)
+        self.app.add_exception_handler(ModelNotFound, model_not_found_handler)
+        self.app.add_exception_handler(ModelNotReady, model_not_ready_handler)
+        self.app.add_exception_handler(
+            NotImplementedError, not_implemented_error_handler
+        )
+        self.app.add_exception_handler(Exception, generic_exception_handler)
 
 
 class UvicornServer:
     def __init__(
         self,
+        app: FastAPI,
         http_port: int,
-        sockets: List[socket.socket],
         data_plane: DataPlane,
         model_repository_extension,
-        enable_docs_url,
         log_config: Optional[Union[str, Dict]] = None,
         access_log_format: Optional[str] = None,
+        workers: int = 1,
     ):
         super().__init__()
-        self.sockets = sockets
-        rest_server = RESTServer(
-            data_plane, model_repository_extension, enable_docs_url
-        )
-        app = rest_server.create_application()
+        rest_server = RESTServer(app, data_plane, model_repository_extension)
+        rest_server.create_application()
         app.add_middleware(
             TimingMiddleware,
             client=PrintTimings(),
             metric_namer=StarletteScopeToName(prefix="kserve.io", starlette_app=app),
         )
         self.cfg = uvicorn.Config(
-            app=app,
+            app="kserve.model_server:app",
```

Comment:
This import string style is required for using `--workers` and `--reload` options. https://www.uvicorn.org/deployment/#running-programmatically